kafka/consumer.go

```diff
@@ -528,20 +581,27 @@ func (c *consumer) consumeTopicPartition(
 			}
 			last = i
 		}
-		// Only commit the records when at least a record has been processed when
-		// AtLeastOnceDeliveryType is set.
+		// Commit the last record offset when one or more records are processed
+		// and the delivery guarantee is set to AtLeastOnceDeliveryType.
+		// NOTE(marclop) When MaxProcessorConcurrency is greater than 1, commits
+		// may happen out of order, which would rewind the consumer. This should
+		// be fixed when AtLeastOnceDelivery is ready to be used.
```

Comment:
I don't think I understand the change or what needs fixing - could someone please clarify?
Isn't ALOD expected to rewind in case there are errors processing some data? What's new?

Reply:
After deliberating with my pillow and some more testing, the `MaxProcessorConcurrency` flag doesn't appear to impact throughput. So I've removed the config flag and hardcoded the `errgroup.Group` limit to 1.

This means there's no perceivable change in execution from the event processing PoV, so this is a backwards-compatible change.